src/cljam/io/vcf/util/validator.clj

```diff
@@ -0,0 +1,124 @@
+(ns cljam.io.vcf.util.validator
+  (:require [cljam.io.vcf.util :as vcf-util]))
+
+(defn- valid-ref? [s]
+  (or (nil? s)
+      (every? #{\A \T \G \C \N
+                \a \t \g \c \n} s)))
+
+(defn- valid-qual? [v] (or (nil? v)
+                           (float? v)
+                           (integer? v)))
+
+(defn- valid-filter? [filt]
+  (or (nil? filt)
+      (= filt :PASS)
+      (and (sequential? filt)
+           (every? keyword? filt))))
+
+(defn- check-base-records [variant meta-included-contig? meta-included-format?]
+  (cond-> nil
+    (some #{\< \> \, \  \tab} (:chr variant))
+    (assoc :chr "must not contain whitespaces commas or angle brackets.")
```

Comment:
If multiple messages for the same key can be proposed at the same time, it should be a sequence.
```suggestion
    (update :chr (fnil conj []) "must not contain whitespaces commas or angle brackets.")
```